src/ensembl/production/dbcopy/views.py

```diff
@@ -61,104 +60,57 @@ def requestjob_checks_warning(request):
-            else:
-                # if any filter (or no filter == all dbs on src) or no match -> union both
-                src_db_set = src_db_set_filter.union(src_db_set_match)
-            logger.info("initial src_db_set %s", src_db_set)
-            excl_db_set_filter = get_database_set(hostname=hostname, port=port,
-                                                  name_filter=src_excl_filter,
-                                                  excluded_schemas=get_excluded_schemas()) if src_excl_filter else set()
-            excl_db_set_match = get_database_set(hostname=hostname, port=port,
-                                                 name_matches=src_excl_match,
-                                                 excluded_schemas=get_excluded_schemas()) if src_excl_match else set()
-            excl_db_set = excl_db_set_filter.union(excl_db_set_match)
-            src_db_set = src_db_set.difference(excl_db_set)
-            logger.info("exc_db_set %s", excl_db_set)
+            src_db_set = get_database_set(hostname=hostname, port=port,
+                                          incl_filters=src_incl_filters,
+                                          skip_filters=src_skip_db_set)
             logger.info("result_db_set %s", src_db_set)
-            if len(src_db_set) == 0 and any(
-                    fil for fil in (src_name_match, src_name_filter, excl_db_set_match, excl_db_set_filter)):
+            if (not src_db_set) and (src_incl_filters or src_skip_filters):
                 # only raise error if no match, but only if any filter specified
-                raise ValueError("No db matching incl. [%s %s] / excl. [%s %s] " % (src_name_filter, src_name_match,
-                                                                                    src_excl_filter, src_excl_match))
+                raise ValueError("No db matching incl. %s / excl. %s " % (src_incl_filters, src_skip_filters))
         except ValueError as e:
             ajax_vars.update({'dberrors': {hostname: [str(e)]}})
-            return HttpResponse(json.dumps(ajax_vars),
-                                status=400,
-                                content_type='application/json')
+            return HttpResponse(json.dumps(ajax_vars), status=400, content_type='application/json')
 
         # 2. For each target:
         #   Retrieve all dbnames which match tgt_db_name
         #   Diff with src_dbames
-        tgt_name_filter, target_name_match = get_filter_match(request.POST.getlist('tgt_db_name', []))
-        logger.debug("src_db_set %s", src_db_set)
-        logger.debug("target_db_set_match %s", target_name_match)
-        tgt_db_set_match = src_db_set
-        # TODO manage tgt_name_filter
-        if target_name_match:
-            tgt_db_set_match = set(target_name_match)
-            logger.debug("Updated src_db_set with renamed targets %s", target_name_match)
+        tgt_db_names = set(request.POST.getlist('tgt_db_name', [])) or src_db_set
```

Comment:
Python question, if default is `[]` then is the `or` called at all?

Reply:
Is called when `request.POST.getlist('tgt_db_name', [])` returns an empty list. From Python docs:

> In the context of Boolean operations, and also when expressions are used by control flow statements, the following values are interpreted as false: `False`, `None`, numeric zero of all types, and empty strings and containers (including strings, tuples, lists, dictionaries, sets and frozensets). All other values are interpreted as true. User-defined objects can customize their truth value by providing a `__bool__()` method.

https://docs.python.org/3/reference/expressions.html#booleans